pkg/snowflake/materialized_view.go

```diff
@@ -0,0 +1,208 @@
+
+	if vb.db != "" && vb.schema != "" {
+		n.WriteString(fmt.Sprintf(`"%v"."%v".`, vb.db, vb.schema))
+	}
+
+	if vb.db != "" && vb.schema == "" {
+		n.WriteString(fmt.Sprintf(`"%v"..`, vb.db))
+	}
+
+	if vb.db == "" && vb.schema != "" {
+		n.WriteString(fmt.Sprintf(`"%v".`, vb.schema))
+	}
+
+	n.WriteString(fmt.Sprintf(`"%v"`, vb.name))
+
+	return n.String()
+}
+
+// WithComment adds a comment to the MaterializedViewBuilder
+func (vb *MaterializedViewBuilder) WithComment(c string) *MaterializedViewBuilder {
+	vb.comment = c
+	return vb
+}
+
+// WithDB adds the name of the database to the MaterializedViewBuilder
+func (vb *MaterializedViewBuilder) WithDB(db string) *MaterializedViewBuilder {
+	vb.db = db
+	return vb
+}
+
+// WithReplace adds the "OR REPLACE" option to the MaterializedViewBuilder
+func (vb *MaterializedViewBuilder) WithReplace() *MaterializedViewBuilder {
+	vb.replace = true
+	return vb
+}
+
+// WithSchema adds the name of the schema to the MaterializedViewBuilder
+func (vb *MaterializedViewBuilder) WithSchema(s string) *MaterializedViewBuilder {
+	vb.schema = s
+	return vb
+}
+
+// WithSecure sets the secure boolean to true
+// [Snowflake Reference](https://docs.snowflake.net/manuals/user-guide/views-secure.html)
+func (vb *MaterializedViewBuilder) WithSecure() *MaterializedViewBuilder {
+	vb.secure = true
+	return vb
+}
+
+// WithStatement adds the SQL statement to be used for the view
+func (vb *MaterializedViewBuilder) WithStatement(s string) *MaterializedViewBuilder {
+	vb.statement = s
+	return vb
+}
+
+// WithClusterBy adds the cluster keys to be used for the view
+func (vb *MaterializedViewBuilder) WithClusterBy(s []interface{}) *MaterializedViewBuilder {
+	vb.clusterBy = s
+	return vb
+}
+
+// View returns a pointer to a Builder that abstracts the DDL operations for a view.
+//
+// Supported DDL operations are:
+//   - CREATE MATERIALIZED VIEW
+//   - ALTER MATERIALIZED VIEW
+//   - DROP MATERIALIZED VIEW
+//   - SHOW MATERIALIZED VIEWS
+//   - DESCRIBE MATERIALIZED VIEW
+//
+// [Snowflake Reference](https://docs.snowflake.com/en/sql-reference/ddl-table.html#materialized-view-management)
+func MaterializedView(name string) *MaterializedViewBuilder {
+	return &MaterializedViewBuilder{
+		name: name,
+	}
+}
+
+// Create returns the SQL query that will create a new view.
+func (vb *MaterializedViewBuilder) Create() string {
+	var q strings.Builder
+
+	q.WriteString("CREATE")
+
+	if vb.replace {
+		q.WriteString(" OR REPLACE")
+	}
+
+	if vb.secure {
+		q.WriteString(" SECURE")
+	}
+
+	q.WriteString(fmt.Sprintf(` MATERIALIZED VIEW %v`, vb.QualifiedName()))
+
+	if vb.comment != "" {
+		q.WriteString(fmt.Sprintf(" COMMENT = '%v'", EscapeString(vb.comment)))
+	}
+
+	if len(vb.clusterBy) != 0 {
+		var s []string
+		for _, e := range vb.clusterBy {
+			s = append(s, e.(string))
+		}
+		q.WriteString(fmt.Sprintf(" CLUSTER BY (%v)", strings.Join(s, ",")))
+	}
+
+	q.WriteString(fmt.Sprintf(" AS %v", vb.statement))
```

Comment:
This might be a long shot, but I think possibly the statement needs to contain a `QualifiedName` as well when you pass in the table. In your test, you specify the `schema.table`.